node/src/actors/chain_manager/mining.rs

```diff
@@ -313,6 +324,36 @@ impl ChainManager {
                     }
                 })
                 .flatten()
+                // Refrain from trying to resolve any more requests if we have already hit the limit
+                // of retrievals per epoch.
+                .and_then(move |vrf_proof| {
+                    let local_retrieval_count = cloned_retrieval_count.load(Relaxed);
+
+                    if local_retrieval_count + additional_retrieval_count > maximum_retrieval_count {
```

Comment:
This operation can overflow a `u16`. To fix, cast everything to `u32`, or change the order of the operations:
```suggestion
                    if local_retrieval_count > maximum_retrieval_count - additional_retrieval_count {
```

Reply:
This is still not fixed? You can also used `checked_add` and return `true` on overflow